Accessors to go between SequencingRead <-> HostFilteredSequencingRead

ttung opened the pull request:
### Description
This makes it easier to walk between objects.

Depends on #85 

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
None.

erinstargazer commented on src/py/aspen/database/models/host_filtering.py:
```diff
@@ -14,6 +21,16 @@ class HostFilteredSequencingRead(Entity):
+    def sequencing_read(self) -> SequencingRead:
+        """The raw gisaid dump this processed gisaid dump was generated from."""
+        # this import has to be here for circular dependencies reasons. :(
+        from .sequences import SequencingRead
```
Is this expensive to import or does it cause a loop (and if it does create a loop why isn't that a problem when `TYPE_CHECKING` is set)? I'm curious about the choice to only import at the top of the file if `TYPE_CHECKING` is true, but then import it here since it's used by this property as opposed to just importing that at the top level as usual.

ttung replied:
It does cause a loop.  mypy is a two-pass compiler, so it doesn't have loop issues, but cpython is one-pass.

erinstargazer reviewed (approved):
Just had a clarification question but otherwise looks good.

Commit:
Accessors to go between SequencingRead <-> HostFilteredSequencingRead

### Description
This makes it easier to walk between objects.

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
None.
